src/model/analysis.ts

```diff
@@ -23,6 +23,11 @@ import {ResolvedUrl} from './url';
 import {Warning} from './warning';
 
 
+export type Result<G, B> = {
+  successful: true,
+  value: G
+}|{successful: false, value: B};
```

Comment:
what about `|{successful: false, error: B}`?

Reply:
Is that suboptimal for v8? Or are the different types for `value` enough to give success and failure different shapes?

Reply:
I wouldn't worry too much about V8 for this API. It's relatively low-frequency, and I've learned that the exact meaning of "call-site" wrt mono/poly/megamorphic isn't what you'd expect. 

The call-site is not when a value is passed to functions, but only when properties are accessed. Since this return type is usually going to be gated by a condition, the accesses to .error vs .value are probably going to remain mono-morphic. The check on .successful might go polymorphic, but again, I think that level of VM optimization is a bit much for this API.